components/remote_settings/src/context.rs

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde_json::{json, Map, Value};
+
+/// Remote settings context object
+///
+/// This is used to filter the records returned. We always fetch all `records` from the
+/// remote-settings storage. Some records could have a `filter_expression`.  If this is passed in
+/// and the record has a `filter_expression`, then only returns where the expression is true will
+/// be returned.
+///
+/// See https://remote-settings.readthedocs.io/en/latest/target-filters.html for details.
+#[derive(Debug, Clone, Default, uniffi::Record)]
+pub struct RemoteSettingsContext {
+    /// The delivery channel of the application (e.g "nightly")
+    #[uniffi(default = None)]
+    pub channel: Option<String>,
+    /// User visible version string (e.g. "1.0.3")
+    #[uniffi(default = None)]
+    pub app_version: Option<String>,
+    /// The locale of the application during initialization (e.g. "es-ES")
+    #[uniffi(default = None)]
+    pub locale: Option<String>,
+    /// The name of the operating system (e.g. "Android", "iOS", "Darwin", "Windows")
+    #[uniffi(default = None)]
+    pub os: Option<String>,
+    /// Form-factor of the device ("phone", "tablet", or "desktop")
+    #[uniffi(default = None)]
+    pub form_factor: Option<String>,
+    /// Country of the user
+    #[uniffi(default = None)]
+    pub country: Option<String>,
+}
+
+impl RemoteSettingsContext {
+    /// Convert this into the `env` value for the remote settings JEXL filter
+    ///
+    /// https://remote-settings.readthedocs.io/en/latest/target-filters.html
+    pub(crate) fn into_env(self) -> Value {
+        let mut v = Map::new();
+        v.insert("channel".to_string(), self.channel.into());
+        if let Some(version) = self.app_version {
+            v.insert("version".to_string(), version.into());
+        }
+        if let Some(locale) = self.locale {
+            v.insert("locale".to_string(), locale.into());
+        }
+        if let Some(os) = self.os {
+            v.insert(
+                "appinfo".to_string(),
+                json!({
+                    "OS": os,
+                }),
+            );
+        }
+        if let Some(form_factor) = self.form_factor {
+            v.insert("form_factor".to_string(), form_factor.into());
```

Comment:
Can we use `formFactor`? Otherwise it would be the only field that is not camelcase.
@cbguder changed it too on Mars https://github.com/mozilla-services/mars/commit/fe8a6cceb207c6bc2bc553e589e4394ad64b9393
([Slack](https://mozilla.enterprise.slack.com/team/U0614EVEZCM))

Reply:
Good question.  @0c0w3 What do you think?  What's currently on remote-settings for the geo-expansion project?

Reply:
Right -- Mathieu, Can, and I [just discussed this](https://mozilla.slack.com/archives/CBGR8AW6R/p1742551041620309), so we're all on the same page here. I think right now the data in RS still uses snake case but Can said he would update it sometime this week. Desktop still passes in snake case too -- we'll need to make a desktop patch for that and this PR (as part of the vendor that includes this PR probably).